crates/sncast/src/starknet_commands/script.rs

```diff
@@ -381,42 +383,28 @@ fn warn_if_sncast_std_not_compatible(scarb_metadata: &Metadata) -> Result<()> {
     Ok(())
 }
 
-fn compile_script(path_to_scarb_toml: Option<Utf8PathBuf>) -> Result<Utf8PathBuf> {
-    let scripts_manifest_path = path_to_scarb_toml.unwrap_or_else(|| {
-        get_scarb_manifest()
-            .context("Failed to retrieve manifest path from scarb")
-            .unwrap()
-    });
-    ensure!(
-        scripts_manifest_path.exists(),
-        "The path = {scripts_manifest_path} does not exist"
-    );
-
-    ScarbCommand::new_with_stdio()
-        .arg("build")
-        .manifest_path(&scripts_manifest_path)
-        .run()
-        .context("failed to compile script with scarb")?;
+fn ensure_lib_artifact(
```

Comment:
I don't know what this function does looking at the name.

Reply:
It's supposed to ensure that script artifacts are present in artifacts map, basically what having `[lib] (...)` in Scarb.toml would do. Should I rename to `ensure_script_artifact` or something similar?

Reply:
this looks like a case for an assert

maybe `assert!(script_lib_artifact_exists(metadata, package_metadata, artifacts)?)`

Reply:
From what I can see, the hash map this function returns is not used anywhere

Reply:
> this looks like a case for an assert
maybe assert!(script_lib_artifact_exists(metadata, package_metadata, artifacts)?)

Not 100% sure, as this function is supposed to add this, not check if this exists. Maybe just `inject_lib_artifact` would be better name then?

> From what I can see, the hash map this function returns is not used anywhere

Added assignment in newest commit, to properly indicate that the run is using it

Reply:
`ensure` suggests it is an assert, but it seems like this function returns something. That's where I am lost.

Reply:
English is nnot my native language, but I'm pretty sure ensure != assert. 

Ensure means to make certain something will happen (ensure_lib_artifact = make sure lib artifact will be in Map, make it happen)

Assert means to state a fact of belief confidently and forcefully (assert_lib_artefact = lib artifact has to be there, otherwise the program needs to fail)

So, it makes sense to me that `ensure_lib_artifact` returns something, as it does not asserts its existence but rather makes sure it's there.

Reply:
As discussed offline, renamed it to `inject_lib_artifact` in https://github.com/foundry-rs/starknet-foundry/pull/1565/commits/ba406e7a45dc22a5ba3f4f276a023957b31f6fb6